engine/runtime/src/main/scala/org/enso/compiler/codegen/IrToTruffle.scala

```diff
@@ -1765,7 +1783,7 @@ class IrToTruffle(
               argumentExpression,
               section,
               displayName,
-              true,
+              subjectToInstrumentation,
```

Comment:
This is the important change. A `ClosureRootNode` is created which inherits `subjectToInstrumentation` flag from outer lambda. The rest of the changes in `IrToTruffle` are only needed to propagate the flag here.